openssl/src/crypto/pkey.rs

```diff
@@ -69,6 +70,22 @@ impl PKey {
         }
     }
 
+    /// Reads private key from PEM, takes ownership of handle
```

Comment:
PKey isn't for private keys only, is it?
At least from the codenbase it looks like it can also hold just a public key:

    pub struct PKey {
        evp: *mut ffi::EVP_PKEY,
        parts: Parts,
    }
    enum Parts {
        Neither,
        Public,
        Both
    }

and e.g. `X509_get_pubkey()` also returns a `EVP_PKEY*`